app/services/editions/db/CollectionsQueries.scala

```diff
@@ -106,19 +106,19 @@ trait CollectionsQueries {
 
     // At the moment we don't do partial updates so simply delete all of them and reinsert.
     sql"""
-          DELETE FROM articles WHERE collection_id = ${collection.id}
+          DELETE FROM cards WHERE collection_id = ${collection.id}
     """.execute().apply()
 
-    collection.items.zipWithIndex.foreach { case (article, index) =>
-      val addedOn = EditionsDB.dateTimeFromMillis(article.addedOn)
+    collection.items.zipWithIndex.foreach { case (card, index) =>
+      val addedOn = EditionsDB.dateTimeFromMillis(card.addedOn)
       sql"""
-          INSERT INTO articles (
+          INSERT INTO cards (
           collection_id,
-          page_code,
+          id,
           index,
           added_on,
           metadata
-          ) VALUES (${collection.id}, ${article.pageCode}, $index, $addedOn, ${article.metadata.map(m => Json.toJson(m).toString)}::JSONB)
+          ) VALUES (${collection.id}, ${card.id}, $index, $addedOn, ${card.metadata.map(m => Json.toJson(m).toString)}::JSONB)
        """.execute().apply()
```

Comment:
Can we do this in a DB transaction block (`BEGIN` / `END`?)

Reply:
This is already the case IIRC, everything within `localTx` is [in a single transaction](https://scalikejdbc.org/documentation/transaction.html)

Confirmed by turning on statement logging (separate PR incoming):

<details>
<summary>Logs</summary>

```
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.177 UTC [55] LOG:  execute <unnamed>: BEGIN
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.178 UTC [55] LOG:  execute <unnamed>: 
2024-04-23 18:14:00 facia-tool-postgres  |            UPDATE collections
2024-04-23 18:14:00 facia-tool-postgres  |            SET is_hidden = $1,
2024-04-23 18:14:00 facia-tool-postgres  |                updated_on = $2,
2024-04-23 18:14:00 facia-tool-postgres  |                updated_by = $3,
2024-04-23 18:14:00 facia-tool-postgres  |                updated_email = $4
2024-04-23 18:14:00 facia-tool-postgres  |            WHERE id = $5
2024-04-23 18:14:00 facia-tool-postgres  |          
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.178 UTC [55] DETAIL:  parameters: $1 = 'f', $2 = '2024-04-23 18:14:00.159+01', $3 = 'Jonathon Herbert', $4 = 'jonathon.herbert@guardian.co.uk', $5 = 'c42e8c1b-1f2e-4b71-af8c-e046ba8c2391'
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.182 UTC [55] LOG:  execute <unnamed>: 
2024-04-23 18:14:00 facia-tool-postgres  |                DELETE FROM cards WHERE collection_id = $1
2024-04-23 18:14:00 facia-tool-postgres  |          
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.182 UTC [55] DETAIL:  parameters: $1 = 'c42e8c1b-1f2e-4b71-af8c-e046ba8c2391'
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.185 UTC [55] LOG:  execute <unnamed>: 
2024-04-23 18:14:00 facia-tool-postgres  |                INSERT INTO cards (
2024-04-23 18:14:00 facia-tool-postgres  |                collection_id,
2024-04-23 18:14:00 facia-tool-postgres  |                id,
2024-04-23 18:14:00 facia-tool-postgres  |                index,
2024-04-23 18:14:00 facia-tool-postgres  |                added_on,
2024-04-23 18:14:00 facia-tool-postgres  |                metadata
2024-04-23 18:14:00 facia-tool-postgres  |                ) VALUES ($1, $2, $3, $4, $5::JSONB)
2024-04-23 18:14:00 facia-tool-postgres  |             
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.185 UTC [55] DETAIL:  parameters: $1 = 'c42e8c1b-1f2e-4b71-af8c-e046ba8c2391', $2 = '14096538', $3 = '0', $4 = '2024-04-23 18:12:01.111+01', $5 = '{"customKicker":"US news","mediaType":"useArticleTrail"}'
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.194 UTC [55] LOG:  execute <unnamed>: 
2024-04-23 18:14:00 facia-tool-postgres  |                INSERT INTO cards (
2024-04-23 18:14:00 facia-tool-postgres  |                collection_id,
2024-04-23 18:14:00 facia-tool-postgres  |                id,
2024-04-23 18:14:00 facia-tool-postgres  |                index,
2024-04-23 18:14:00 facia-tool-postgres  |                added_on,
2024-04-23 18:14:00 facia-tool-postgres  |                metadata
2024-04-23 18:14:00 facia-tool-postgres  |                ) VALUES ($1, $2, $3, $4, $5::JSONB)
2024-04-23 18:14:00 facia-tool-postgres  |             
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.194 UTC [55] DETAIL:  parameters: $1 = 'c42e8c1b-1f2e-4b71-af8c-e046ba8c2391', $2 = '14053006', $3 = '1', $4 = '2024-04-23 18:13:59.647+01', $5 = '{"customKicker":"Other lives","mediaType":"useArticleTrail"}'
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.198 UTC [55] LOG:  execute <unnamed>: 
2024-04-23 18:14:00 facia-tool-postgres  |            SELECT
2024-04-23 18:14:00 facia-tool-postgres  |              collections.id,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.front_id,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.index,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.name,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.is_hidden,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.metadata,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.updated_on,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.updated_by,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.updated_email,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.prefill,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.path_type,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.content_prefill_window_start,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.content_prefill_window_end,
2024-04-23 18:14:00 facia-tool-postgres  |              fronts.is_special,
2024-04-23 18:14:00 facia-tool-postgres  | 
2024-04-23 18:14:00 facia-tool-postgres  |              cards.collection_id AS cards_collection_id,
2024-04-23 18:14:00 facia-tool-postgres  |              cards.id            AS cards_id,
2024-04-23 18:14:00 facia-tool-postgres  |              cards.card_type     AS cards_card_type,
2024-04-23 18:14:00 facia-tool-postgres  |              cards.index         AS cards_index,
2024-04-23 18:14:00 facia-tool-postgres  |              cards.added_on      AS cards_added_on,
2024-04-23 18:14:00 facia-tool-postgres  |              cards.metadata      AS cards_metadata
2024-04-23 18:14:00 facia-tool-postgres  | 
2024-04-23 18:14:00 facia-tool-postgres  |            FROM collections
2024-04-23 18:14:00 facia-tool-postgres  |            LEFT JOIN cards ON (cards.collection_id = collections.id)
2024-04-23 18:14:00 facia-tool-postgres  |            LEFT JOIN fronts ON (collections.front_id = fronts.id)
2024-04-23 18:14:00 facia-tool-postgres  |            WHERE collections.id = $1
2024-04-23 18:14:00 facia-tool-postgres  |            
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.198 UTC [55] DETAIL:  parameters: $1 = 'c42e8c1b-1f2e-4b71-af8c-e046ba8c2391'
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.199 UTC [55] LOG:  execute S_1: COMMIT
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.201 UTC [55] LOG:  execute <unnamed>: 
2024-04-23 18:14:00 facia-tool-postgres  |          SELECT edition_issues.id AS id
2024-04-23 18:14:00 facia-tool-postgres  |          FROM edition_issues
2024-04-23 18:14:00 facia-tool-postgres  |          INNER JOIN fronts ON (fronts.issue_id = edition_issues.id)
2024-04-23 18:14:00 facia-tool-postgres  |          INNER JOIN collections ON (collections.front_id = fronts.id)
2024-04-23 18:14:00 facia-tool-postgres  |          WHERE collections.id = $1
2024-04-23 18:14:00 facia-tool-postgres  |          
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.201 UTC [55] DETAIL:  parameters: $1 = 'c42e8c1b-1f2e-4b71-af8c-e046ba8c2391'
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.202 UTC [55] LOG:  execute <unnamed>: 
2024-04-23 18:14:00 facia-tool-postgres  |            SELECT
2024-04-23 18:14:00 facia-tool-postgres  |              edition_issues.id,
2024-04-23 18:14:00 facia-tool-postgres  |              edition_issues.name,
2024-04-23 18:14:00 facia-tool-postgres  |              edition_issues.timezone_id,
2024-04-23 18:14:00 facia-tool-postgres  |              edition_issues.issue_date,
2024-04-23 18:14:00 facia-tool-postgres  |              edition_issues.created_on,
2024-04-23 18:14:00 facia-tool-postgres  |              edition_issues.created_by,
2024-04-23 18:14:00 facia-tool-postgres  |              edition_issues.created_email,
2024-04-23 18:14:00 facia-tool-postgres  |              edition_issues.launched_on,
2024-04-23 18:14:00 facia-tool-postgres  |              edition_issues.launched_by,
2024-04-23 18:14:00 facia-tool-postgres  |              edition_issues.launched_email,
2024-04-23 18:14:00 facia-tool-postgres  | 
2024-04-23 18:14:00 facia-tool-postgres  |              fronts.id            AS fronts_id,
2024-04-23 18:14:00 facia-tool-postgres  |              fronts.issue_id      AS fronts_issue_id,
2024-04-23 18:14:00 facia-tool-postgres  |              fronts.index         AS fronts_index,
2024-04-23 18:14:00 facia-tool-postgres  |              fronts.name          AS fronts_name,
2024-04-23 18:14:00 facia-tool-postgres  |              fronts.is_special    AS fronts_is_special,
2024-04-23 18:14:00 facia-tool-postgres  |              fronts.is_hidden     AS fronts_is_hidden,
2024-04-23 18:14:00 facia-tool-postgres  |              fronts.metadata      AS fronts_metadata,
2024-04-23 18:14:00 facia-tool-postgres  |              fronts.updated_on    AS fronts_updated_on,
2024-04-23 18:14:00 facia-tool-postgres  |              fronts.updated_by    AS fronts_updated_by,
2024-04-23 18:14:00 facia-tool-postgres  |              fronts.updated_email AS fronts_updated_email,
2024-04-23 18:14:00 facia-tool-postgres  | 
2024-04-23 18:14:00 facia-tool-postgres  |              collections.id            AS collections_id,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.front_id      AS collections_front_id,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.index         AS collections_index,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.name          AS collections_name,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.is_hidden     AS collections_is_hidden,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.metadata      AS collections_metadata,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.updated_on    AS collections_updated_on,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.updated_by    AS collections_updated_by,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.updated_email AS collections_updated_email,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.prefill       AS collections_prefill,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.path_type     AS collections_path_type,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.content_prefill_window_start       AS collections_content_prefill_window_start,
2024-04-23 18:14:00 facia-tool-postgres  |              collections.content_prefill_window_end         AS collections_content_prefill_window_end,
2024-04-23 18:14:00 facia-tool-postgres  | 
2024-04-23 18:14:00 facia-tool-postgres  |              cards.collection_id AS cards_collection_id,
2024-04-23 18:14:00 facia-tool-postgres  |              cards.id            AS cards_id,
2024-04-23 18:14:00 facia-tool-postgres  |              cards.card_type     AS cards_card_type,
2024-04-23 18:14:00 facia-tool-postgres  |              cards.index         AS cards_index,
2024-04-23 18:14:00 facia-tool-postgres  |              cards.added_on      AS cards_added_on,
2024-04-23 18:14:00 facia-tool-postgres  |              cards.metadata      AS cards_metadata
2024-04-23 18:14:00 facia-tool-postgres  | 
2024-04-23 18:14:00 facia-tool-postgres  |            FROM edition_issues
2024-04-23 18:14:00 facia-tool-postgres  |            LEFT JOIN fronts ON (fronts.issue_id = edition_issues.id)
2024-04-23 18:14:00 facia-tool-postgres  |            LEFT JOIN collections ON (collections.front_id = fronts.id)
2024-04-23 18:14:00 facia-tool-postgres  |            LEFT JOIN cards ON (cards.collection_id = collections.id)
2024-04-23 18:14:00 facia-tool-postgres  |            WHERE edition_issues.id = $1
2024-04-23 18:14:00 facia-tool-postgres  |            
2024-04-23 18:14:00 facia-tool-postgres  | 2024-04-23 17:14:00.202 UTC [55] DETAIL:  parameters: $1 = '041293dd-8f95-455d-aa94-0fcc382f4b3b'
```

</details>

Reply:
https://github.com/guardian/facia-tool/pull/1564

Reply:
😎